tests/ui/sanitizer/cfi-complex-receiver.rs

```diff
@@ -2,11 +2,16 @@
 // * Arc<dyn Foo> as for custom receivers
 // * &dyn Bar<T=Baz> for type constraints
 
-//@ needs-sanitizer-cfi
+//@ revisions: cfi kcfi
 // FIXME(#122848) Remove only-linux once OSX CFI binaries work
 //@ only-linux
-//@ compile-flags: --crate-type=bin -Cprefer-dynamic=off -Clto -Zsanitizer=cfi
-//@ compile-flags: -C target-feature=-crt-static -C codegen-units=1 -C opt-level=0
+//@ [cfi] needs-sanitizer-cfi
+//@ [kcfi] needs-sanitizer-kcfi
+//@ compile-flags: -C opt-level=0
```

Comment:
This is not. I think? Why do you need to specify the opt-level is 0, exactly? I know this was previously approved but I didn't notice it was setting the opt-level last time. ^^;

Reply:
`opt-level=0` isn't fully mandatory, and I could remove it from the set. I had it enabled to guard against an overzealous optimizer exploiting accidental (as in a bug from our CFI implementation) undefined behavior to result in a successful call.

Should I remove it?

Reply:
@maurer Probably. We have jobs that run the tests at different optimization levels but that won't override the explicit optimization level set here, so if there's a difference in behavior you won't find out.

Reply:
Done.